java/client/src/main/java/glide/api/models/BaseTransaction.java

```diff
@@ -1231,9 +1234,8 @@ public <ArgType> T rpopCount(@NonNull ArgType key, long count) {
      * @remarks If <code>key</code> does not exist, a new set is created before adding <code>members
      *     </code>.
      */
-    public T sadd(@NonNull String key, @NonNull String[] members) {
-        ArgsArray commandArgs = buildArgs(ArrayUtils.addFirst(members, key));
-        protobufTransaction.addCommands(buildCommand(SAdd, commandArgs));
+    public <ArgType> T sadd(@NonNull ArgType key, @NonNull ArgType[] members) {
```

Comment:
You're not protected from `ArgType` being bool or `Map` or whatever else.
If you decide to parametrize it instead of duplicating, please:
1. Add what is `ArgType` to the docs
2. Protect `ArgType` from being something else except `String` or `GlideString`
3. Add tests for this ^

ref https://github.com/aws/glide-for-redis/pull/1691#discussion_r1660457758

Reply:
What would be the best way to protect `ArgType` without calling `instnceof` for two unrelated classes (`String` and `GlideString`) ?

Reply:
Probably, no way. What is bad in `instanceof`?